smetrics/src/main/scala/com/evolutiongaming/smetrics/CollectionHelper.scala

```diff
@@ -0,0 +1,38 @@
+package com.evolutiongaming.smetrics
+
+import cats.Semigroupal
+import cats.implicits._
+
+object CollectionHelper {
+
+  implicit class CombinationGenerator[A](values: List[List[A]]) {
+
+    private val combinator: (List[A], List[A]) => List[List[A]] =
+      Semigroupal.map2[List, A, A, List[A]](_, _)((value1, value2) => List(value1, value2))
+
+    /*
+      Generates all possible combinations for transmitted lists.
+
+      Examples:
+        List.empty -> List.empty
+        List(1,2,3)) -> List(List(1), List(2), List(3))
+        List(1,2), List(3,4)) -> List(List(1,3), List(1,4), List(2,3), List(2,4))
+        List(List(1, 2), List(3,4), List(5,6)) -> List(List(1, 3, 5), List(1, 3, 6), List(1, 4, 5), List(1, 4, 6), List(2, 3, 5), List(2, 3, 6), List(2, 4, 5), List(2, 4, 6))
+     */
+    def generateCombinations: List[List[A]] = {
```

Comment:
does `List.permutations` the same?

Reply:
could you please explain why this method is needed at all?

Reply:
`does List.permutations the same?` -  I'll check

`could you please explain why this method is needed at all?` - Sure, we need to create all possible combinations of our labels. 

For example .
Metric{label1=[v1,v2], label2=[v1,v2], label3=[v1,v2]}.
v1,v1,v1
v1,v2,v1
....
v2,v2,v2
....